### PR TITLE
Solve batch issue with auto executor

### LINKED ISSUE #37: NotImplementedError when getting a job result for version 1.1.4
Hi,

I have been using `submitit` a lot recently to submit my jobs on a SLURM-based HPC. It has been very good to me, allowing me to get rid of an extra API writing, so thanks for this project.

When I installed version 1.1.4, I got an error whose stacktrace is as follows:

```python
Traceback (most recent call last):
  File "three_d.py", line 19, in <module>
    train_eval_grid(
  File "/gpfsdswork/projects/rech/hih/uap69lx/submission-scripts/jean_zay/submitit/general_submissions.py", line 63, in train_eval_grid
    run_ids = [job.result() for job in jobs]
  File "/gpfsdswork/projects/rech/hih/uap69lx/submission-scripts/jean_zay/submitit/general_submissions.py", line 63, in <listcomp>
    run_ids = [job.result() for job in jobs]
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 266, in result
    r = self.results()
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 284, in results
    self.wait()
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 387, in wait
    while not self.done():
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 424, in done
    if self.watcher.is_done(self.job_id, mode="force" if force_check else "standard"):
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 100, in is_done
    state = self.get_state(job_id, mode=mode)
  File "/linkhome/rech/gencea10/uap69lx/.conda/envs/dis-mri-recon/lib/python3.8/site-packages/submitit/core/core.py", line 53, in get_state
    raise NotImplementedError
NotImplementedError
```

I didn't get the chance to work on a minimal reproducible example yet, but I can tell you that this happens in 1.1.4 and doesn't happen in 1.1.3.
I can point you in the meantime to the failing line in my public code: [here](https://github.com/zaccharieramzi/submission-scripts/blob/master/jean_zay/submitit/general_submissions.py#L63).

### PR DESCRIPTION
**Pull request by jrapin**
Closes #37

### CODE REVIEW

File: submitit/core/core.py [hunk 1]
```diff
@@ -585,11 +585,15 @@ def batch(self) -> tp.Iterator[None]:
...
-            job: Job[R] = self.job_class.__new__(self.job_class)  # empty shell
+            # ugly hack for AutoExecutor class which is known at runtime
+            cls = self.job_class if self.job_class is not Job else self._executor.job_class  # type: ignore
+            job: Job[R] = cls.__new__(cls)  # empty shell
             self._delayed_batch.append((job, ds))
-            return job
         else:
-            return self._internal_process_submissions([ds])[0]
+            job = self._internal_process_submissions([ds])[0]
+        if type(job) is Job:  # pylint: disable=unidiomatic-typecheck
```
Comment (gwenzek) [inline submitit/core/core.py:594]: I'd like a small comment here of why we need the exact type match

File: submitit/auto/auto.py [hunk 2]
```diff
@@ -171,7 +171,12 @@ def _internal_update_parameters(self, **kwargs: Any) -> None:
                 )
                 continue
         if invalid:
-            invalid.append(f"Known executors: {', '.join(executors.keys())}")
+            invalid.extend(
```
Comment (gwenzek) [inline submitit/auto/auto.py:174]: this seems unrelated ? did you intend to push it ? looks good otherwise

**Review by gwenzek — CHANGES_REQUESTED** [review]
Please also bump the version counter, we want a quick release here

File: submitit/auto/auto.py (hunk 2 above)
Comment (jrapin) [reply submitit/auto/auto.py:174]: Not related indeed, but I thought about improving the error message since I kept getting "unknown cluster 'tasks'" which I found unhelpful. The issue was that I was using `tasks_per_job` instead of `tasks_per_node`

File: submitit/core/core.py (hunk 1 above)
Comment (jrapin) [reply submitit/core/core.py:594]: Not sure what I can add given the line below states: `raise RuntimeError("Executors should never return a base Job class (implementation issue)")` 
is there a different way to check that it is a subclass but not the class ?

Comment (gwenzek) [reply submitit/core/core.py:594]: good for me, I didn't see the RuntimeError in my review

**Review by gwenzek — APPROVED** [review]
